src/Customer.php

```diff
@@ -106,7 +106,14 @@ public static function findByExternalId($externalId)
         if (gettype($externalId) == 'string') {
             $externalId = ['external_id' => $externalId];
         }
-        return static::all($externalId)->entries->first();
+
+        $response = static::all($externalId);
+
+        if ($response->first()) {
```

Comment:
You need to check $response here isn't null, otherwise you're just shuffling the error around.